### PR TITLE
AWS RHEL Scaleup

### PR DESCRIPTION
**Pull request by yunjiang29**
Adding RHEL scale-up on AWS clusters with following configurations:
* enabled cluster proxy
* disconnected & private cluster

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision
+  steps:
+  - chain: cucushift-installer-rehearse-aws-ipi-disconnected-private-provision
+  - ref: workers-rhel-aws-provision
+  - ref: workers-rhel-pre-hook-sshtunnel
+  - chain: workers-rhel
+  env:
+  - name: USE_INTERNAL_SSH_BASTION
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml:9]: How about renaming it to `USE_SSH_BASTION_POD`?

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision
+  steps:
+  - chain: cucushift-installer-rehearse-aws-ipi-disconnected-private-provision
+  - ref: workers-rhel-aws-provision
+  - ref: workers-rhel-pre-hook-sshtunnel
+  - chain: workers-rhel
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml:7]: Per yesterday's meeting, suggest to append `ref: mirror-images-enable-qe-catalogsource` here.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/cco-manual-security-token-service/provision/cucushift-installer-rehearse-aws-ipi-proxy-cco-manual-security-token-service-provision-chain.yaml [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+  - ref: aws-provision-vpc-disconnected
+  - chain: aws-provision-bastionhost
+  - ref: ipi-conf
+  - ref: ipi-conf-aws-custom-vpc
+  - ref: ipi-conf-proxy
+  - ref: ipi-conf-aws
+  - ref: ipi-install-monitoringpvc
   - ref: ipi-conf-manual-creds
   - ref: ipi-conf-aws-oidc-creds-provision
   - chain: ipi-install
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/cco-manual-security-token-service/provision/cucushift-installer-rehearse-aws-ipi-proxy-cco-manual-security-token-service-provision-chain.yaml:13]: Per yesterday's meeting, suggest to append ref: mirror-images-enable-qe-catalogsource here.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/provision/cucushift-installer-rehearse-aws-ipi-proxy-provision-chain.yaml [hunk 4]
```diff
@@ -1,7 +1,13 @@
...
   steps:
-  - chain: ipi-conf-aws-proxy
+  - ref: aws-provision-vpc-disconnected
+  - chain: aws-provision-bastionhost
+  - ref: ipi-conf
+  - ref: ipi-conf-aws-custom-vpc
+  - ref: ipi-conf-proxy
+  - ref: ipi-conf-aws
+  - ref: ipi-install-monitoringpvc
   - chain: ipi-install
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/provision/cucushift-installer-rehearse-aws-ipi-proxy-provision-chain.yaml:11]: Per yesterday's meeting, suggest to append ref: mirror-images-enable-qe-catalogsource here.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/techpreview/provision/cucushift-installer-rehearse-aws-ipi-proxy-techpreview-provision-chain.yaml [hunk 5]
```diff
@@ -1,7 +1,13 @@
...
-  - chain: ipi-conf-aws-proxy
+  - ref: aws-provision-vpc-disconnected
+  - chain: aws-provision-bastionhost
+  - ref: ipi-conf
+  - ref: ipi-conf-aws-custom-vpc
+  - ref: ipi-conf-proxy
+  - ref: ipi-conf-aws
+  - ref: ipi-install-monitoringpvc
   - ref: ipi-conf-techpreview
   - chain: ipi-install
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/proxy/techpreview/provision/cucushift-installer-rehearse-aws-ipi-proxy-techpreview-provision-chain.yaml:12]: Per yesterday's meeting, suggest to append ref: mirror-images-enable-qe-catalogsource here.

File: ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh [hunk 6]
```diff
@@ -26,6 +31,18 @@ if ! whoami &> /dev/null; then
     fi
 fi
 
+#Get bastion info
+if [[ "${USE_INTERNAL_SSH_BASTION}" == "yes" ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh:35]: Seem like a bastion is a MUST for rhel scaleup. 

So maybe we can drop `USE_INTERNAL_SSH_BASTION` here, and do some pre-check for bastion.
```
if [ -f "${SHARED_DIR}/bastion_public_address" ]; then
  BASTION_SSH_USER=$(cat "${SHARED_DIR}/bastion_ssh_user")
  BASTION_PUBLIC_ADDRESS=$(cat "${SHARED_DIR}/bastion_public_address")
else
  BASTION_SSH_USER="core"
  BASTION_PUBLIC_ADDRESS=$(oc get service ssh-bastion --namespace=test-ssh-bastion --output=jsonpath='{.status.loadBalancer.ingress[0].hostname}')
if 

if [ -z "${BASTION_PUBLIC_ADDRES}" ]; then
  echo "Did not find bastion public address"
  exit 1
else
  export BASTION_SSH_USER
  export BASTION_PUBLIC_ADDRESS
fi
```

Comment (yunjiang29) [reply ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh:35]: good point, will update it.

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml:9]: addressed in https://github.com/openshift/release/pull/29782#discussion_r909222081

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml (hunk 2 above)
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/disconnected/private/workers-rhcos-rhel8/provision/cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8-provision-chain.yaml:7]: already added in `cucushift-installer-rehearse-aws-ipi-disconnected-private-provision`

File: ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh [hunk 7]
```diff
@@ -26,6 +31,22 @@ if ! whoami &> /dev/null; then
     fi
 fi
 
+if [ -f "${SHARED_DIR}/bastion_public_address" ]; then
+  BASTION_SSH_USER=$(cat "${SHARED_DIR}/bastion_ssh_user")
+  BASTION_PUBLIC_ADDRESS=$(cat "${SHARED_DIR}/bastion_public_address")
+else
+  BASTION_SSH_USER="core"
+  BASTION_PUBLIC_ADDRESS=$(oc get service ssh-bastion --namespace=test-ssh-bastion --output=jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+if
```
Comment (jianlinliu) [inline ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh:40]: Typo, should be `fi`

File: ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh [hunk 8]
```diff
@@ -26,6 +31,22 @@ if ! whoami &> /dev/null; then
...
 
+if [ -f "${SHARED_DIR}/bastion_public_address" ]; then
+  BASTION_SSH_USER=$(cat "${SHARED_DIR}/bastion_ssh_user")
+  BASTION_PUBLIC_ADDRESS=$(cat "${SHARED_DIR}/bastion_public_address")
+else
+  BASTION_SSH_USER="core"
+  BASTION_PUBLIC_ADDRESS=$(oc get service ssh-bastion --namespace=test-ssh-bastion --output=jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+if 
+
+if [ -z "${BASTION_PUBLIC_ADDRESS}" ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh:42]: to be more restrict, suggest append `|| [ -z "${BASTION_SSH_USER}" ]` there.

**Comment by yunjiang29** [comment]
Hello @patrickdillon, Can you review and approve this PR? Thanks!

/assign @patrickdillon

**Pull request merged**